controllers/relayer_controller.go

```diff
@@ -0,0 +1,224 @@
+//+kubebuilder:rbac:groups=terra.terra-rebels.org,resources=relayers/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=terra.terra-rebels.org,resources=relayers/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+// TODO(user): Modify the Reconcile function to compare the state specified by
+// the Relayer object against the actual cluster state, and then
+// perform operations to make the cluster state reflect the state specified by
+// the user.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.13.0/pkg/reconcile
+func (r *RelayerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	logger.Info("Reconciling Relayer object")
+
+	relayer := &terrav1alpha1.Relayer{}
+	err := r.Client.Get(ctx, req.NamespacedName, relayer)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+
+		return ctrl.Result{}, err
+	}
+
+	pod := newPodForRelayer(relayer)
+
+	if err := controllerutil.SetControllerReference(relayer, pod, r.Scheme); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	foundPod := &corev1.Pod{}
+	err = r.Client.Get(context.TODO(), types.NamespacedName{Name: pod.Name, Namespace: pod.Namespace}, foundPod)
+
+	if err != nil && errors.IsNotFound(err) {
+		logger.Info("Creating a new Pod", "Pod.Namespace", pod.Namespace, "Pod.Name", pod.Name)
+
+		err = r.Client.Create(context.TODO(), pod)
+
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+
+		return ctrl.Result{}, nil
+	} else if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *RelayerReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&terrav1alpha1.Relayer{}).
+		Complete(r)
+}
+
+func newPodForRelayer(cr *terrav1alpha1.Relayer) *corev1.Pod {
+	labels := map[string]string{
+		"app": cr.Name,
+	}
+
+	srcPort := "transfer"
+	if cr.Spec.SrcPort != "" {
+		srcPort = cr.Spec.SrcPort
+	}
+
+	dstPort := "transfer"
+	if cr.Spec.DstPort != "" {
+		dstPort = cr.Spec.DstPort
+	}
+
+	icsVersion := "ics20-1"
+	if cr.Spec.ICSVersion != "" {
+		icsVersion = cr.Spec.ICSVersion
+	}
+
+	firstMinGasAmount := "0"
+	if cr.Spec.FirstNetwork.MinGasAmount != "" {
+		firstMinGasAmount = cr.Spec.FirstNetwork.MinGasAmount
+	}
+
+	secondMinGasAmount := "0"
+	if cr.Spec.SecondNetwork.MinGasAmount != "" {
+		secondMinGasAmount = cr.Spec.SecondNetwork.MinGasAmount
+	}
+
+	envVars := []corev1.EnvVar{
+		{
+			Name:  "SRC_PORT",
+			Value: srcPort,
+		},
+		{
+			Name:  "DST_PORT",
+			Value: dstPort,
+		},
+		{
+			Name:  "VERSION",
+			Value: icsVersion,
+		},
+		{
+			Name:  "FIRST_NETWORK_NAME",
```

Comment:
I really dont like FIRST / SECOND naming conventions. SOURCE / DESTINATION might not truly describe the full duplex nature of the relationship, but as such we could also call it PEER_A and PEER_B?